satpy/utils.py

```diff
@@ -281,66 +281,82 @@ def atmospheric_path_length_correction(data, cos_zen, limit=88., max_sza=95.):
     return data * corr
 
 
-def get_satpos(dataset):
+def get_satpos(
+        data_arr: xr.DataArray,
+        preference: Optional[str] = None,
```

Comment:
Could this preference be also fetched from the central satpy configuration if not provided?

Reply:
Yes! Sorry I missed this. I commented on slack. This is exactly the idea I had, but I think the preference should be use case specific. Meaning, the config option would be for controlling sensor angle generation and how it uses this versus a global/generic "whenever satellite position is requested prefer 'X'".

Reply:
I still think there should be a central config for this that can be undefined by default, and still allow this flag to be present. But this is good enough for now.

Reply:
I'm not completely against this idea, but I'm not sure what purpose it serves. Users don't typically use `get_satpos`, only other Satpy components do. Those satpy components (ex. angle generation) should be the things with the config options for the preference.

Reply:
As discussed on slack, what @mraspaud is going discussing something like a group of configuration parameters or a "profile" where you can say `satpy.config.set_group("image_optimizations")` and the various time or satellite positions or other parameters would be set to specific values that would be allow for the fastest generation of images. Of course I just made up the name for that parameter group, but hopefully the idea is clear. This would be a big feature and is not part of this PR, but I wanted to mention it here as something that would be affected by this type of configuration grouping.